apps/src/code-studio/components/AssetRow.jsx

```diff
@@ -5,15 +5,82 @@ import AssetThumbnail from './AssetThumbnail';
 import i18n from '@cdo/locale';
 import firehoseClient from '@cdo/apps/lib/util/firehose';
 import color from '@cdo/apps/util/color';
+import Dialog, {Body} from '@cdo/apps/templates/Dialog';
+import Button from '../../templates/Button';
+import DialogFooter from '../../templates/teacherDashboard/DialogFooter';
 
 const styles = {
   deleteWarning: {
     paddingLeft: '34px',
     textAlign: 'left',
     color: color.red
+  },
+  leftAlign: {
+    textAlign: 'left'
   }
 };
 
+class ConfirmImageDeleteDialog extends React.Component {
+  static propTypes = {
+    name: PropTypes.string.isRequired,
+    usage: PropTypes.number,
+    onCancel: PropTypes.func.isRequired,
+    handleDelete: PropTypes.func.isRequired,
+    // For logging purposes
+    studyGroup: PropTypes.string.isRequired,
+    projectId: PropTypes.string.isRequired,
+    elementId: PropTypes.string
+  };
+
+  onConfirmDelete = () => {
+    firehoseClient.putRecord({
+      study: 'delete-asset',
+      study_group: this.props.studyGroup,
+      event: 'confirm-referenced',
+      project_id: this.props.projectId,
+      data_json: JSON.stringify({
+        assetName: this.props.name,
+        elementId: this.props.elementId,
+        usage: this.props.usage
+      })
+    });
+    this.props.handleDelete();
+  };
+
+  render() {
+    return (
+      <Dialog
+        title={i18n.warning()}
+        isOpen
+        handleClose={this.props.onCancel}
+        style={styles.leftAlign}
+      >
+        <Body>
+          <p>
+            {i18n.deleteUsedImage({
+              name: this.props.name,
+              value: this.props.usage,
+              places: this.props.usage > 1 ? i18n.places() : i18n.place()
```

Comment:
this will help in many languages, however you should be aware that some languages have more than two plural forms. For example, Polish has different word endings for cardinalities of 1, 2-4, and 5+. Irish has 5 different plural forms! http://docs.translatehouse.org/projects/localization-guide/en/latest/l10n/pluralforms.html

Ideally we'd use a translations framework that recognizes plurality and generates multiple translatable strings when a number is provided as a placeholder. CC @Hamms @dju90 

until then, I think this is a good improvement :)

Reply:
oh, we actually do already support this!

In this case, what you'll want to do is create the `deletedUsedImage` string like so:

```
"{name} is used in {value, plural, one {1 place} other {# places}} in this app. ..."
```

Then just pass in `name` and `value`. Translators will be able to add translations for `place` and `places`, and - if they know how - also add translations for whatever other pluralization categories their language uses.

Reply:
![shockedowl](https://user-images.githubusercontent.com/2959170/54787648-9e7d8480-4be9-11e9-809c-3d88f44e073c.gif)
That's awesome!
When I use i18n.deleteUsedImage, do I just need to pass the `name` and `value` parameter to get that^ to work?

Reply:
Right?!

And yep! You can see more info about this syntax in the [official docs](https://messageformat.github.io/messageformat/)

Reply:
Wow, so cool! :D